lib/spack/env/cc

```diff
@@ -222,59 +314,66 @@ dtags_to_strip="${SPACK_DTAGS_TO_STRIP}"
 linker_arg="${SPACK_LINKER_ARG}"
 
 # Set up rpath variable according to language.
-eval rpath=\$SPACK_${comp}_RPATH_ARG
+rpath="ERROR: RPATH WAS NOT SET"
+eval "rpath=\$SPACK_${comp}_RPATH_ARG"
 
 # Dump the mode and exit if the command is dump-mode.
-if [[ $SPACK_TEST_COMMAND == dump-mode ]]; then
+if [ "$SPACK_TEST_COMMAND" = "dump-mode" ]; then
     echo "$mode"
     exit
 fi
 
 # Check that at least one of the real commands was actually selected,
 # otherwise we don't know what to execute.
-if [[ -z $command ]]; then
+if [ -z "$command" ]; then
     die "ERROR: Compiler '$SPACK_COMPILER_SPEC' does not support compiling $language programs."
```

Comment:
Unless I'm mistaken, this is dead code, since the script should error out earlier in the `case` statement at:
```bash
*)
    die "Unknown compiler: $command"`
```
so I would recommend moving this error message up there.

Reply:
I think @cosmicexplorer is correct that this code block is unreachable.

Reply:
I meant to dig into this but lost it in the chain of 51 suggestions -- sorry.

This *is* reachable and I've seen it in builds. It's reachable if, say, `SPACK_FC` is not set but `SPACK_CC` is.  Those variables aren't required to be set, and users aren't required to put a fortran compiler in `compilers.yaml`.  This `if` handles the case where we inadvertently try to build some Fortran without a Fortran compiler.

I added a commit clarifying this case and tried to make the `cc` error messages better, too.